lib/http_server.js

```diff
@@ -446,10 +446,34 @@ ZettaHttpServer.prototype.proxyToPeer = function(env, next) {
     });
 
     res.statusCode = response.statusCode;
-    var buf = [];
+
     if(!opts.pipe) {
-      response.on('data', function(d) {
-        buf += d;  
+      var body = null;
+      var buf = [];
+      var len = 0;
+
+      response.on('readable', function() {
+        var chunk;
+
+        while ((chunk = response.read()) != null) {
+          buf.push(chunk);
+          len += chunk.length;
+        }
+
+        if (!buf.length) {
+          return;
+        }
+
+        if (buf.length && Buffer.isBuffer(buf[0])) {
```

Comment:
Might be a leftover from upgrading to Streams2.  We should probably kill it everywhere.  Or better yet, replace it with something a little safer.  That's an OOM attack waiting to happen.  Most solutions (Apache, IIS) will begin to buffer to disk until a defined maximum is reached.